### DISCUSSION TITLE
Rebuilding an Electron Desktop app with Tauri

### DISCUSSION THREAD

**Discussion by Morsmalleo** [tauri-apps/tauri#4873] [Q&A] [answered]
I am currently maintaining a revival of a popular electron project, this project uses the following node modules since I've obtained it...
- `angular`
- `angular-route`
- `fomantic-ui`
- `fs-extra` (but the regular built in JS `fs` module also works fine)
- `geoip-lite`
- `node-homedir` (`homedir` also works)
- `socket.io`

With these node modules being used to run the application, I'm wondering if Tauri provides anything similar to these modules? Because I'm planning to move the project to Tauri for better memory efficiency as well as better cross platform operations.

**Comment by FabianLars** [comment, marked as answer]
So the only real constraint you have in a Tauri app is that you can't use npm packages that need a nodejs runtime _at runtime_.
This means that angular, angular-route and fomantic-ui will keep working.
- fs-extra: Alternatives in Tauri are Tauri's built-in [fs module](https://tauri.app/v1/api/js/modules/fs) and our [fs-extra plugin](https://github.com/tauri-apps/tauri-plugin-fs-extra), the latter being super bare bones right now, but we're open for feature requests
- geoip-lite: That's a tricky one, you probably won't get around a Rust based solution here, unless you know of a npm package that works in a browser
- (node-)homedir: https://tauri.app/v1/api/js/modules/path#homedir
- socket.io: **If you just use its client you're fine**, if you use the server package than it will be a tricky one too. You either need Rust or a nodejs [sidecar](https://tauri.app/v1/guides/building/sidecar) (compiled via pkg or similar)

**Reply by Morsmalleo** [reply]
I did find this for geoip-lite but not sure if it's what I'm looking for 

https://github.com/oschwald/maxminddb-rust

As for socket.io, yes it's the server package obtained with `npm install socket.io@version.number` it's used to connect the server to the client which uses `socket.io-client java`, I did find this but it's very old

https://crates.io/crates/socket-io

As for the rest of the modules I need, they will all work, except for `angular, angular-route & fomantic-ui` yes?
